test/parallel/test-runner-cli.js

```diff
@@ -28,11 +28,10 @@ const testFixtures = fixtures.path('test-runner');
   assert.strictEqual(child.signal, null);
   assert.strictEqual(child.stderr.toString(), '');
   const stdout = child.stdout.toString();
-  assert.match(stdout, /ok 1 - .+index\.test\.js/);
-  assert.match(stdout, /not ok 2 - .+random\.test\.mjs/);
-  assert.match(stdout, /not ok 1 - this should fail/);
+  assert.match(stdout, /ok 1 - this should pass/);
+  assert.match(stdout, /not ok 2 - this should fail/);
   assert.match(stdout, /ok 3 - .+subdir.+subdir_test\.js/);
```

Comment:
It feels inconsistent to me that we report empty files and files that change their exit code but don't have test failures like this. Maybe we should lean on the existing warning diagnostic mechanism that we already use to convey this information since they aren't actually tests.

Reply:
I saw quite a few parsers that relay on a failing test to know that the entire suite needs to fail, I dont feel like diagnostics is enough in this case

Reply:
we kind of relay on that as well https://github.com/nodejs/node/blob/806a516851f906c6cf0ca7b45dbcf499855b4c54/lib/internal/main/test_runner.js#L26

Reply:
So we essentially support four types of files:

1. Files containing tests that exit with code 0 (all passing).
2. Files containing tests that exit with a non-zero code (one or more failing).
3. Files containing no tests that exit with code 0.
4. Files containing no tests that exit with a non-zero code.

I think we have a good idea of how to handle cases 1 and 2.

Cases 3 and 4 are the edge cases that you called out in the original PR description. After thinking about this some more, I think I agree with the approach that you've taken here for two reasons.

1. They are edge cases that should not happen often and aren't the end of the world.
2. I've said before (and still believe) that we shouldn't force people to use `node:test` in order to use the test runner. If we relied on printing a diagnostic here, then the user experience would suffer for those people.